main.cpp

```diff
@@ -228,346 +305,521 @@ struct University
     float pulseWidth = 0.3f; 
-    //octave (n times the original frequency )
     int octave = 2; 
+    double loadedROM;
+    double storedROM;
+    bool playButton;
 
-    //generate audio signals
-    void generateSignal(Oscillator Square);
-    //load samples from ROM
-    void loadROMSamples(std::string selectStorageDevice = "SD",
+    double generateSignal(Oscillator Square, double x); 
+    double loadROMSamples(std::string selectStorageDevice = "SD",
                         bool isAudioFormat = true);
-    //playback samples from ROM
-    void playbackROMSamples(bool anyKeyPressed = false);
+    void playbackROMSamples(double x, bool anyKeyPressed = false);
 };
 
-/*
-Thing 8) filters
-5 properties:
-    1) gain (float)
-    2) bandwidth (float)
-    3) frequency (float)
-    4) type (std::string)
-    5) drive (float)
-3 things it can do:
-    1) boost or cut frequencies on a signal
-    2) overDrive the signal
-    3) give a better sonority to an instrument 
- */
+double Oscillator::generateSignal(Oscillator Square, double x)
+{
+    if (Square.waveformShape == "Square")
+    {
+        x = 0.5 * (frequency); //This is a dummy example
+    }
+    return x; 
+}
+
+double Oscillator::loadROMSamples(std::string selectStorageDevice,
+                                  bool isAudioFormat)
+{
+    if (selectStorageDevice == "SD")
+    {
+        if(isAudioFormat == true)
+        loadedROM = true;
+    }
+    return storedROM;
+}
+
+void Oscillator::playbackROMSamples(double x, bool anyKeyPressed)
+{
+    do 
+    {
+        x = storedROM + 1;
+    }
+    while (anyKeyPressed == true);
+}
 
 struct Filters 
 {
-    //gain [dB] pow(10.0, (gain / 20.0))
-    float gain = 0.0f; 
-    // bandwidth [0 -10] 0.7=1 Oct
-    float bandwidth = 0.7f; 
-    //frequency [Hz] Selected center frequency
-    float frequency = 100.0f; 
-    //type (LP HP LSh HSh BP)
+    double gain; 
+    double bandwidth = 0.7; 
+    double frequency = 100.0; 
     std::string type = "Low Pass"; 
-    //drive (Adds distortion to the signal)
-    float drive = 0.0f; 
-
-    //boost or cut frequencies on a signal
-    void boostCutFreq(Filters LowPass);
-    //overDrive the signal
-    void overDriveSignal(float drive); 
-    //give a better sonority to an instrument 
-    void giveSonority(std::string instrument = "piano");
+    double drive = 0.0; 
+    double pianoIR;
+
+    double boostCutFreq(Filters LowShelf, double x);
+    double overDriveSignal(double drive, double x); 
+    double giveSonority(double x, std::string instrument = "Piano");
 };
 
-/*
-Thing 9) reverb 
-5 properties:
-    1) time (float) 
-    2) mix (float)
-    3) type (std::string)
-    4) pre-delay (float)
-    5) size (float)
-3 things it can do:
-    1) simulate an space to a soun
-    2) give a better sound to an instrument
-    3) give depth to a band
- */
+double Filters::boostCutFreq(Filters LowShelf, double x)
+{
+    LowShelf.gain=-12;
```

Comment:
Do not pass in an instance of `Filter` into a `Filter` member function
You can use the member variables of the calling instance directly inside the member function. 
- [ ] revise